opm/simulators/linalg/bda/Reorder.cpp

```diff
@@ -0,0 +1,372 @@
+    int *resRowsPerIter;
+
+    std::vector <int> rowsToStart;
+
+    // find starting rows: rows that are independent from all rows that come before them.
+    for (thisRow = 0; thisRow < Nb; thisRow++) {
+        if (canBeStarted(thisRow, CSRRowPointers, CSCColIndices, doneRows)) {
+            fromOrder[nextActiveRowIndex] = thisRow;
+            toOrder[thisRow] = nextActiveRowIndex;
+            nextActiveRowIndex++;
+        }
+    }
+    // 'do' compute on all active rows
+    for (iterEnd = 0; iterEnd < nextActiveRowIndex; iterEnd++) {
+        doneRows[fromOrder[iterEnd]] = true;
+    }
+
+    rowsPerIter.emplace_back(nextActiveRowIndex - activeRowIndex);
+
+    while (iterEnd < Nb) {
+        // Go over all rows active from the last iteration, and check which of their neighbours can be activated this iteration
+        for (; activeRowIndex < iterEnd; activeRowIndex++) {
+            thisRow = fromOrder[activeRowIndex];
+
+            for (int i = CSCRowPointers[thisRow]; i < CSCRowPointers[thisRow + 1]; i++) {
+                int thatRow = CSCColIndices[i];
+
+                if (canBeStarted(thatRow, CSRRowPointers, CSRColIndices, doneRows)) {
+                    rowsToStart.emplace_back(thatRow);
+                }
+            }
+        }
+        // 'do' compute on all active rows
+        for (unsigned int i = 0; i < rowsToStart.size(); i++) {
+            thisRow = rowsToStart[i];
+            if (!doneRows[thisRow]) {
+                doneRows[thisRow] = true;
+                fromOrder[nextActiveRowIndex] = thisRow;
+                toOrder[thisRow] = nextActiveRowIndex;
+                nextActiveRowIndex++;
+            }
+        }
+        iterEnd = nextActiveRowIndex;
+        rowsPerIter.emplace_back(nextActiveRowIndex - activeRowIndex);
+    }
+    // Crop the rowsPerIter array to it minimum size.
+    int numColors = rowsPerIter.size();
+    resRowsPerIter = new int[numColors];
+    for (int i = 0; i < numColors; i++) {
+        resRowsPerIter[i] = rowsPerIter[i];
+    }
+
+    *iters = rowsPerIter.size();
+
+    return resRowsPerIter;
+}
+
+/* Perform the complete graph coloring algorithm on a matrix. Return an array with the amount of nodes per color.*/
+
+int* findGraphColoring(int *colIndices, int *rowPointers, int Nb, int maxRowsPerColor, int maxColsPerColor, int *numColors, int *toOrder, int* fromOrder) {
+    std::vector<int> rowColor;
+    rowColor.resize(Nb);
+    int *rowsPerColor = new int[MAX_COLORS];
+
+    colorBlockedNodes(Nb, rowPointers, colIndices, rowColor, maxRowsPerColor, maxColsPerColor);
+
+    colorsToReordering(Nb, rowColor, toOrder, fromOrder, rowsPerColor);
+
+    *numColors = MAX_COLORS;
+    while (rowsPerColor[*numColors - 1] == 0) {
+        *numColors = *numColors - 1;
+    }
+
+    return rowsPerColor;
+}
+
+// based on the scipy package from python, scipy/sparse/sparsetools/csr.h on github
+// input : matrix A via Avals, Acols, Arows, Nb
+// output: matrix B via Bvals, Bcols, Brows
```

Comment:
Which one is CSR and which one is CSC? One cannot judge by the variable names. Maybe indicate with a name which array is start pointer? E.g. for CSR use rowStart and col, for CSC use colStart and row. Currently this is a bit confusing and makes it hard for others to understand the code.

Reply:
Addressed in 38c58bf. The CSC nonzeroes are not needed in Flow, only the pattern is converted.